File: app/src/main/java/com/cazimir/relaxoo/ui/favorites/FavoritesViewModel.java
package com.cazimir.relaxoo.ui.favorites;

import android.util.Log;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.cazimir.relaxoo.R;
import com.cazimir.relaxoo.model.SavedCombo;
import com.cazimir.relaxoo.model.Sound;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class FavoritesViewModel extends ViewModel {

  private static final String TAG = "FavoritesViewModel";

  private MutableLiveData<List<SavedCombo>> savedCombos = new MutableLiveData<>();

  public FavoritesViewModel() {
    fetchFavorites();
  }

  public MutableLiveData<List<SavedCombo>> savedCombosLive() {
    return savedCombos;
  }

  private void fetchFavorites() {

    Log.d(TAG, "fetchFavorites: called");

    SavedCombo savedCombo1 =
        new SavedCombo.Builder()
            .withName("Test combo 1")
            .withSounds(
                Arrays.asList(
                    Sound.newSound("Sound1", R.drawable.ic_windy, R.raw.sound1, false, 0.5f),
                    Sound.newSound("Sound2", R.drawable.ic_windy, R.raw.sound3, false, 0.5f)))
            .build();

    SavedCombo savedCombo2 =
            new SavedCombo.Builder()
                    .withName("Test combo 2")
                    .withSounds(
                            Arrays.asList(
                                    Sound.newSound("Sound1", R.drawable.ic_windy, R.raw.sound1, false, 0.5f),
                                    Sound.newSound("Sound2", R.drawable.ic_windy, R.raw.sound3, false, 0.5f)))
                    .build();


    ArrayList<SavedCombo> list = new ArrayList<>(Arrays.asList(savedCombo1, savedCombo2));

    savedCombos.setValue(list);
  }
}

File: app/src/main/java/com/cazimir/relaxoo/ui/favorites/FavoritesViewModel.kt
package com.cazimir.relaxoo.ui.favorites

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.cazimir.relaxoo.model.ListOfSavedCombos
import com.cazimir.relaxoo.model.SavedCombo

class FavoritesViewModel : ViewModel() {

    companion object {
        private const val TAG = "FavoritesViewModel"
    }

    private val _savedCombos = MutableLiveData<ListOfSavedCombos>()

    lateinit var repository: IFavoritesRepository

    fun savedCombosLive(): MutableLiveData<ListOfSavedCombos> {
        return _savedCombos
    }

    fun fetchFavorites() {

        val favorites = repository.getFavorites()

        if (favorites == null) {
            val emptyListOfSavedCombos = ListOfSavedCombos(mutableListOf())
            _savedCombos.value = emptyListOfSavedCombos
            return
        }

        _savedCombos.value = favorites
    }

    fun deleteCombo(position: Int) {
        val currentSavedCombos = _savedCombos.value
        val currentList = currentSavedCombos?.savedComboList?.toMutableList()
        currentList?.removeAt(position)
        _savedCombos.value = ListOfSavedCombos(currentList)
        repository.addUpdatedList(ListOfSavedCombos(currentList))
    }

    fun addFavorite(comboToBeAdded: SavedCombo) {
        // get current object with list and return another one with savedCombo added
        val currentSavedCombos = _savedCombos.value
        val currentList = currentSavedCombos?.savedComboList?.toMutableList()
        currentList?.add(comboToBeAdded)
        _savedCombos.value = ListOfSavedCombos(currentList)
        repository.addUpdatedList(ListOfSavedCombos(currentList))
    }
}
